credoai/artifacts.py

```diff
@@ -379,7 +379,7 @@ def __init__(
         self.name = name
         self.config = {}
         assert model is not None or model_config is not None
-        if model is not None:
+        if model is not None and model_config is None:
```

Comment:
What was this change for? The reason I had set it up like so was to make sure we are always inferring what we can from the model, and only updating what was explicitly asked for.